Doc/c-api/time.rst

```diff
@@ -0,0 +1,132 @@
+.. highlight:: c
+
+PyTime C API
+============
+
+.. versionadded:: 3.13
+
+PyTime API
+----------
+
+The PyTime_t API is written to use timestamp and timeout values stored in
+various formats and to read clocks.
+
+The :c:type:`PyTime_t` type is an integer to support directly common arithmetic
+operations such as ``t1 + t2``.
+
+The PyTime_t API supports a resolution of ``1`` nanosecond. The
+:c:type:`PyTime_t` type is signed to support negative timestamps. The supported
+range is around [-292.3 years; +292.3 years]. Using the Unix epoch (January
+1st, 1970), the supported date range is around [1677-09-21; 2262-04-11].
+
+Time formats:
+
+* Seconds.
+* Seconds as a floating pointer number (C :c:expr:`double`).
+* Milliseconds (10\ :sup:`-3` seconds).
+* Microseconds (10\ :sup:`-6` seconds).
+* 100 nanoseconds (10\ :sup:`-7` seconds), used on Windows.
+* Nanoseconds (10\ :sup:`-9` seconds).
+* :c:expr:`timeval` structure, 1 microsecond (10\ :sup:`-6` seconds).
+* :c:expr:`timespec` structure, 1 nanosecond (10\ :sup:`-9` seconds).
+
+Integer overflows are detected and raise :exc:`OverflowError`. Conversion to a
+resolution larger than 1 nanosecond is rounded correctly with the requested
+rounding mode. Available rounding modes:
+
+* Round towards minus infinity (-inf). For example, used to read a clock.
+* Round towards infinity (+inf). For example, used for timeout to wait "at
+  least" N seconds.
+* Round to nearest with ties going to nearest even integer. For example, used
+  to round from a Python float.
+* Round away from zero. For example, used for timeout.
+
+Some functions clamp the result in the range [PyTime_MIN; PyTime_MAX]. The
+caller doesn't have to handle errors and so doesn't need to hold the GIL to
+handle exceptions. For example, ``_PyTime_Add(t1, t2)`` computes ``t1+t2`` and
```

Comment:
`_PyTime_Add` is mentioned, but it does not look like it's part of the public API.